### PR TITLE
Add support for Google Cloud ARM instances

### PR DESCRIPTION
**Pull request by francescolavra**
This change set adds support for running Nanos images on Google Cloud aarch64 instances, which boot via UEFI, implement ACPI and are equipped with NVME disks and gVNIC network adapters.

### CODE REVIEW

File: src/drivers/gve.c [hunk 1]
```diff
@@ -0,0 +1,855 @@
...
+            p->len = p->tot_len = length;
+            pbuf_ref(p);
+        } else {
+            gve_debug("RX packet copy");
+            p = pbuf_alloc(PBUF_RAW, length, PBUF_RAM);
+            if (p) {
+                pbuf_take(p, payload, length);
+            } else {
+                msg_err("failed to allocate pbuf\n");
+                continue;
```
Comment (sanderssj) [inline src/drivers/gve.c:563]: Need lwip_unlock() before continue

**Review by sanderssj — COMMENTED** [review]
LGTM

File: src/drivers/gve.c (hunk 1 above)
Comment (francescolavra) [reply src/drivers/gve.c:564]: Good catch, I added lwip_unlock() now

**Review by wjhun — APPROVED** [review]
One comment about contiguous allocations, otherwise LGTM

File: src/drivers/gve.c [hunk 2]
```diff
@@ -0,0 +1,856 @@
...
+static void gve_deinit_interrupts(gve adapter)
+{
+    pci_teardown_msix(adapter->pdev, 1);
+    pci_teardown_msix(adapter->pdev, 2);
+    pci_disable_msix(adapter->pdev);
+}
+
+static void *gve_create_qpl(gve adapter, u16 num_pages, u32 id)
+{
+    void *qpl_base = allocate(adapter->h, num_pages * PAGESIZE);
```
[the remark this answers is not in the document]
Comment (francescolavra) [reply src/drivers/gve.c]: OK, agreed, now the linear backed heap is used to allocate memory for shared/device use.

File: src/drivers/gve.c [hunk 3]
```diff
@@ -0,0 +1,856 @@
...
+{
+    int msix_avail = pci_enable_msix(adapter->pdev);
+    if (msix_avail < 3) /* TX irq, RX irq, management irq */
+        return false;
+    if (pci_setup_msix(adapter->pdev, 2, init_closure(&adapter->mgmt_irq_handler, gve_mgmt_irq),
+                       "gve_mgmt") == INVALID_PHYSICAL)
+        goto error;
+    if (pci_setup_msix(adapter->pdev, 1, init_closure(&adapter->rx.irq_handler, gve_rx_irq),
+                       "gve_rx") == INVALID_PHYSICAL) {
+        pci_teardown_msix(adapter->pdev, 0);
```
[the remark this answers is not in the document]
Comment (francescolavra) [reply src/drivers/gve.c]: Oops, that was a leftover from an initial implementation which used the TX interrupt (at MSI slot 0). Removed

**Pull request merged**